### PR TITLE
Update testing practices documentation

### PR DESCRIPTION
**Pull request by stefaniuk**
This change includes the following:
- Format document according to the Markdown standard
- Update the `API Testing Guidance` section to include basic requirements

### CODE REVIEW

File: practices/testing.md [hunk 1]
```diff
@@ -35,50 +35,55 @@
...
-  
+
 * **Testing should be continually improved**
 
 * **Testing is continuous**
   * Testing is a continuous activity, not a phase of delivery
-  * Testing can be performed in [all phases of a product's DevOps cycle](https://danashby.co.uk/2016/10/19/continuous-testing-in-devops/)
-
-
+  * Testing can be performed in [all phases of a product's DevOps cycle](https://danashby.co.uk/2016/10/19/continuous-testing-in-devops/) and is best done as a collaborative exercise
```
Comment (iw-ezequiel) [inline practices/testing.md:44]: Great point - should we reinforce even further by saying it "should" be done collaboratively?

File: practices/testing.md [hunk 2]
```diff
@@ -35,50 +35,55 @@
...
+
+* When building an API start with an implementation of an automated contract test suite first. This can be supported by simple tools like Wiremock and Postman/Newman
+* Use contract tests to support technical and business documentation
+* Provide a web interface for exploratory testing. This usually comes out of the box with tooling that supports the OpenAPI specification. Although, this requires an environment to be set up or to be run on demand locally
 
+### UI Testing Guidance
 
+### Browser/OS Testing Guidance
+
+* Website user interfaces should be tested against an appropriate set of browsers and versions &mdash; in particular, no effort should be spent testing against unsupported browsers or unsupported versions of browsers. See supported browsers for [Staff](https://aalto.digital.nhs.uk/#/document/viewer/8c039de1-eec0-49cd-8af3-a97fed6a8bff?library=5464c07f-daf1-4eee-b9b6-22e6c4dfbbd0) and [Citizens](https://aalto.digital.nhs.uk/#/document/viewer/465e6d1b-f107-49eb-ad25-e72c0299d3a6?library=5464c07f-daf1-4eee-b9b6-22e6c4dfbbd0)
```
Comment (iw-ezequiel) [inline practices/testing.md:76]: Do we have public links for these to keep consistent with all content in the framework being in the open?

Comment (andyblundell) [reply practices/testing.md:76]: Good spot - not sure if we do. Also: we should mention no-closed-links in the contributions page (which we haven't written yet). Let's convert this thread to an issue for later.

File: practices/testing.md (hunk 1 above)
Comment (stefaniuk) [reply practices/testing.md:44]: Done.

**Review by andyblundell — APPROVED** [review]
Looks good to me

**Pull request merged**